src/Nethermind/Nethermind.AccountAbstraction/Source/UserOperationTxSource.cs

```diff
@@ -49,17 +48,17 @@ public class UserOperationTxSource : ITxSource
             _logger = logger;
         }
 
-        public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit)
+        public Transaction? GetTransaction(BlockHeader head, ulong gasLimit)
         {
             IDictionary<Address, HashSet<UInt256>> usedAccessList = new Dictionary<Address, HashSet<UInt256>>();
             IList<UserOperation> userOperationsToInclude = new List<UserOperation>();
             ulong gasUsed = 0;
 
-            IEnumerable<UserOperation> userOperations = 
+            IEnumerable<UserOperation> userOperations =
                 _userOperationPool
                     .GetUserOperations()
-                    .Where(op => op.MaxFeePerGas >= parent.BaseFeePerGas)
-                    .OrderByDescending(op => CalculateUserOperationPremiumGasPrice(op, parent.BaseFeePerGas));
+                    .Where(op => op.MaxFeePerGas >= head.BaseFeePerGas)
```

Comment:
Not sure what GetUserOperations returns but this is potential place for optimization